File: database_comunicator/Models/DTOs/RegisterUser.cs
namespace database_comunicator.Models.DTOs
{
    public class RegisterUser
    {
        public string Email { get; set; } = null!;

        public string Username { get; set; } = null!;

        public string Surname { get; set; } = null!;
        public string OrgName { get; set; } = null!;

        public int? Nip { get; set; }

        public string Street { get; set; } = null!;

        public string City { get; set; } = null!;

        public string PostalCode { get; set; } = null!;
        public string Country { get; set; } = null!;
        public string Password { get; set; } = null!;
        public bool IsOrg { get; set; }
    }
}

File: web/handler_b2b/src/utils/registration/registerUser.js
"use server";

/**
 * Sends request to create new first user.
 * @param  {FormData} formData Contain form data.
 * @param  {string} dbName Database name.
 * @param  {boolean} isOrg Is the account organization type (true) or solo (false).
 * @return {Promise<boolean>}      Result of operation. If true success, otherwise failure.
 */
export default async function createNewRegisteredUser(formData, dbName, isOrg) {
  let nip = formData.get("nip");
  nip = nip === "" ? null : parseInt(nip);
  let data = {
    email: formData.get("email"),
    username: formData.get("name"),
    surname: formData.get("surname"),
    orgName: formData.get("company"),
    nip: nip,
    street: formData.get("street"),
    city: formData.get("city"),
    postalCode: formData.get("postal"),
    country: formData.get("country"),
    password: formData.get("password"),
    isOrg: isOrg === "true",
  };
  try {
    let response = await fetch(
      `${process.env.API_DEST}/${dbName}/Registration/registerUser`,
      {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(data),
      },
    );

    return response.ok;
  } catch {
    console.error("createNewRegisteredUser fetch failed.");
    return false;
  }
}
